### PR TITLE
Likes and views

### PR DESCRIPTION
**Pull request by MalikSaqibIftikhar**
## Description

Fixes #

## Checklist
- [x] Targeted PR against correct branch.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Run `meteor npm run lint`
- [ ] Added an entry to the `CHANGELOG.md` file.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: imports/api/actions/server/methods.js [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+  useDefaultAuth: true,
+  prettyJson: true
+})
+
+Api.addRoute('ping', {authRequired: false}, {
+
+  get: function () {
+
+    return {
+      Code: 200,
```
Comment (faisalnaveedRNS) [inline imports/api/actions/server/methods.js]: remove static code 200

File: imports/api/actions/server/methods.js [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+        Code: 400,
+        Message: "invalid request",
+        Data: null
+      }
+    }
+
+    var result = GetViews(this.urlParams.cookbookId, this.urlParams.recipeId)
+
+    return {
+      Code: 200,
```
Comment (faisalnaveedRNS) [inline imports/api/actions/server/methods.js]: remove static code 200

File: imports/api/actions/server/methods.js [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+        Code: 400,
+        Message: "invalid request",
+        Data: null
+      }
+    }
+
+    var result = ViewNFT(this.urlParams.cookbookId, this.urlParams.recipeId, this.bodyParams.userId)
+
+    return {
+      Code: 200,
```
Comment (faisalnaveedRNS) [inline imports/api/actions/server/methods.js]: remove static code

**Review by faisalnaveedRNS — APPROVED** [review]

File: imports/api/actions/server/methods.js [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+const StatusOk = 200
+const StatusInvalidInput = 400
+
+Api.addRoute('ping', {authRequired: false}, {
+
+  get: function () {
+
+    return {
+      Code: StatusOk,
+      Message: "Api server is up and running!",
```
Comment (faisalnaveedRNS) [inline imports/api/actions/server/methods.js]: put messages in string vars

File: imports/api/actions/server/methods.js [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+
+  },
+
+  //view a specific nft
+  post: function () {
+
+    if ( !Valid(this.urlParams.cookbookId) || !Valid(this.urlParams.recipeId) || !Valid(this.bodyParams.userId) ){
+      return {
+        Code: StatusInvalidInput,
+        Message: "invalid request",
```
Comment (faisalnaveedRNS) [inline imports/api/actions/server/methods.js]: code repetition in message

File: imports/api/actions/server/methods.js [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+
+  },
+
+  //like a specific nft
+  post: function () {
+
+    if ( !Valid(this.urlParams.cookbookId) || !Valid(this.urlParams.recipeId) || !Valid(this.bodyParams.userId) ){
+      return {
+        Code: StatusInvalidInput,
+        Message: "invalid request",
```
Comment (faisalnaveedRNS) [inline imports/api/actions/server/methods.js]: same here

File: imports/api/actions/server/methods.js [hunk 7]
```diff
@@ -0,0 +1,352 @@
...
+
+Api.addRoute('actions/likes/:userId/:cookbookId/:recipeId', {authRequired: false}, {
+
+  //check if the specified user has liked the specified nft or not
+  get: function () {
+
+    if ( !Valid(this.urlParams.cookbookId) || !Valid(this.urlParams.recipeId) || !Valid(this.urlParams.userId) ){
+      return {
+        Code: StatusInvalidInput,
+        Message: "invalid request",
```
Comment (faisalnaveedRNS) [inline imports/api/actions/server/methods.js]: same here

File: imports/api/actions/server/methods.js [hunk 8]
```diff
@@ -0,0 +1,352 @@
...
+        Data: null
+      }
+    }
+
+    var result = GetLikeStatus(this.urlParams.cookbookId, this.urlParams.recipeId, this.urlParams.userId)
+
+    return {
+      Code: StatusOk,
+      Message: "Successful",
```
Comment (faisalnaveedRNS) [inline imports/api/actions/server/methods.js]: same here

File: imports/api/actions/server/methods.js [hunk 9]
```diff
@@ -0,0 +1,352 @@
...
+      return {
+        Code: StatusInvalidInput,
+        Message: "invalid request",
+        Data: null
+      }
+    }
+    var result = ViewNFT(cookbookId, recipeId, userId) 
+    return {
+      Code: StatusOk,
+      Message: "Successful",
```
Comment (faisalnaveedRNS) [inline imports/api/actions/server/methods.js]: same here

File: imports/api/actions/server/methods.js [hunk 10]
```diff
@@ -0,0 +1,352 @@
...
+        Code: StatusInvalidInput,
+        Message: "invalid request",
+        Data: null
+      }
+    }
+    //get likes on this nft
+    var likes = GetLikes(cookbookId, recipeId)
+    return {
+      Code: StatusOk,
+      Message: "Successful",
```
Comment (faisalnaveedRNS) [inline imports/api/actions/server/methods.js]: same here

File: imports/api/actions/server/methods.js [hunk 11]
```diff
@@ -0,0 +1,352 @@
...
+        Code: StatusInvalidInput,
+        Message: "invalid request",
+        Data: null
+      }
+    }
+    //get like status for this user
+    var result = GetLikeStatus(cookbookId, recipeId, userId)
+    return {
+      Code: StatusOk,
+      Message: "Successful",
```
Comment (faisalnaveedRNS) [inline imports/api/actions/server/methods.js]: same here

**Review by faisalnaveedRNS — APPROVED** [review]

**Review by MikeSofaer — APPROVED** [review]

**Pull request merged**